core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -648,6 +654,42 @@ protected int getTotalBlocklets() {
     return sum;
   }
 
+  public Map<String, Long> getRowCount(List<PartitionSpec> partitions,
+      Map<String, Long> blockletToRowCountMap, boolean isUpdateFlow)
+      throws IOException {
+    if (taskSummaryDMStore.getRowCount() == 0) {
+      return new HashMap<>();
+    }
+    // if it has partitioned datamap but there is no partitioned information stored, it means
+    // partitions are dropped so return empty list.
+    if (partitions != null) {
+      if (!validatePartitionInfo(partitions)) {
+        return new HashMap<>();
+      }
+    }
+    Iterator<Map.Entry<String, Integer>> iterator =
+        taskSummaryDMStore.getRowCountMap().entrySet().iterator();
+    if (!isUpdateFlow) {
+      long totalSize = 0;
+      while (iterator.hasNext()) {
+        Map.Entry<String, Integer> next = iterator.next();
+        totalSize += next.getValue();
+      }
+      Long rowCount = blockletToRowCountMap.get("RowCount");
```

Comment:
Better to change "RowCount" to a constant

Reply:
done